crops/crop.go

```diff
@@ -115,3 +124,36 @@ func (c Crop) ComputeConsequences(event interface{}) consequences.Results {
 	r := consequences.Results{IsTable: false, Result: ret}
 	return r
 }
+func (c Crop) computeImpactedCase(e hazards.ArrivalandDurationEvent) float64 {
+	// Determine damage percent based on damage dur curve and event dur
+	perdmg := c.lossFunction.ComputeDamagePercent(e)
+	fmt.Println("damage percent is ", perdmg)
+	croploss := (perdmg / 100) * c.GetValuePerOutputUnit()
+	fmt.Println("Crop loss is : ", croploss)
+	// value added to field before loss by production
+	fmt.Println("total pruduction costs are:", c.productionFunction.productionCostLessHarvest)
+	fmt.Println("fixed costs are: ", c.productionFunction.GetfixedCosts())
+	loss := croploss + c.productionFunction.productionCostLessHarvest
```

Comment:
this doesnt seem quite right - the math should look more like this.

```
        protected virtual void ComputeFloodedDollarDamages(IAgriculturalFloodEvent floodEvent, ref ICropResult result)
        {
            double totalValue = TotalAnnualValue();
			double totalValueLessHarvest = totalValue - ProductionFunction.HarvestCost;
            double exposedvalue = ProductionFunction.ComputeExposedValue(floodEvent);
            double damagePercent = DamageFunction.ComputeDamagePercent(floodEvent);
            double totalCost = ProductionFunction.CumulativeCostsLessHarvest;
            double proRatedValue = exposedvalue / totalCost;
            result.Damage = damagePercent * proRatedValue * totalValueLessHarvest;
        }

Reply:
hhmm yeah missing the prorate piece of this. I'll look into it

Reply:
Also it should be multiply not addition. make sure loss := croploss * c.productionFunction.productionCostlessHarvest

Reply:
oh... I mistakenly thought that cumulateMonthlyCosts(...) takes event info, and was using those fields as such. oops

Reply:
Modified this, but not totally confident in the math. Can you take another look?

Reply:
Reviewed and fixed with @HenryGeorgist